deep_quoridor/src/quoridor_env.py

```diff
@@ -547,3 +592,38 @@ def env(**kwargs):
         return wrappers.CaptureStdoutWrapper(QuoridorEnv(**kwargs))
     else:
         return QuoridorEnv(**kwargs)
+
+
+class StepRewardCalculator:
+    def __init__(self, env: QuoridorEnv = None):
+        self.env = env
+
+    def _get_distances(self):
+        """
+        Gets the current distances to target for both agent and opponent.
+        Returns tuple of (agent_distance, opponent_distance)
+        """
+        agent = self.env.agent_selection
+        agent_distance = self.env.get_distance_to_target(agent)
+
+        opponent = self.env.get_opponent(agent)
+        opponent_distance = self.env.get_distance_to_target(opponent)
+
+        return agent_distance, opponent_distance
+
+    def before_step(self):
+        self.orig_agent_distance, self.orig_opponent_distance = self._get_distances()
+
+    def after_step(self):
+        agent_distance, opponent_distance = self._get_distances()
+
+        # Calculate the reward based on the distance to the goal
+        reward = 0
+        distance_change = (self.orig_agent_distance - agent_distance) - (
+            self.orig_opponent_distance - opponent_distance
+        )
+        if distance_change > 0:
+            reward = 1
+        elif distance_change < 0:
```

Comment:
@muralx @adamantivm @alejandromarcu have any of you tried making the step reward larger depending on the distance change? For instance adding a wall might increase the opponent distance to goal by a lot.